component/config.go

```diff
@@ -160,20 +169,40 @@ func MustNewType(strType string) Type {
 
 // DataType is a special Type that represents the data types supported by the collector. We currently support
 // collecting metrics, traces and logs, this can expand in the future.
-type DataType = Type
-
-func mustNewDataType(strType string) DataType {
-	return MustNewType(strType)
-}
+type DataType string
```

Comment:
I prefer to use a struct here to disallow users to create new DataType's

Reply:
Hmmm... I understand that's a risk.  I've tried to mitigate it with the logic in pipelines/config.go that make sure its a member of signalNameToDataType.  

Do you think its still worth wrapping it in a struct?